pyro/distributions/multivariate_normal.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+import torch
+from torch.autograd import Variable
+
+from pyro.distributions.util import copy_docs_from
+from pyro.distributions.distribution import Distribution
+
+
+def potri_compat(var):
+    return Variable(torch.potri(var.data)) if torch.__version__ < '0.3.0' else torch.potri(var)
+
+
+@copy_docs_from(Distribution)
+class MultivariateNormal(Distribution):
+    """
+    Multivariate normal (Gaussian) distribution.
+
+    A distribution over vectors in which all the elements have a joint
+    Gaussian density.
+
+    :param torch.autograd.Variable loc: Mean. Must be a vector (Variable containing a 1d Tensor).
+    :param torch.autograd.Variable covariance_matrix: Covariance matrix. Must be symmetric and positive semidefinite.
+    :param torch.autograd.Variable scale_tril: The Cholesky decomposition of the covariance matrix. You can pass this
+            instead of `covariance_matrix`.
+    :param use_inverse_for_batch_log: If this is set to true, the torch.inverse function will be used to compute
+        log_pdf. This means that the results of log_pdf can be differentiated with respect to the covariance matrix.
+        Since the gradient of torch.potri is currently not implemented, differentiation of log_pdf wrt. covariance
+        matrix is not possible when using the Cholesky decomposition. Using the Cholesky decomposition is however much
+        faster and therefore enabled by default.
+    :param normalized: If set to `False` the normalization constant is omitted in the results of batch_log_pdf and
+            log_pdf. This might be preferable, as computing the determinant of the covariance matrix might not
+            always be numerically stable. Defaults to `True`.
+    :raises: ValueError if the shape of any parameter is not supported.
+    """
+
+    reparameterized = True
+
+    def __init__(self, loc, covariance_matrix, scale_tril=None, batch_size=None, use_inverse_for_batch_log=False,
```

Comment:
I think the easiest interface to use would be to require one of the args to be specified (as with `Bernoulli`):
```py
def __init__(self, loc, covariance_matrix=None, scale_tril=None, ...):
    if covariance_matrix is None and scale_tril is None:
        raise ValueError('At least one of covariance_matrix or scale_tril must be specified')
    if scale_tril is None:
        covariance_matrix = ...
    elif covariance_matrix is None:
        scale_tril = ...
```

Reply:
Okay, I am going to update that. That will make using the `None`s in the test obsolete too.